src/main/java/htsjdk/samtools/SamReaderFactory.java

```diff
@@ -320,6 +320,10 @@ public SamReader open(final SamInputResource resource) {
         /** Attempts to detect whether the file is an SRA accessioned file. If SRA support is not available, returns false. */
         private boolean isSra(final File sourceFile) {
             try {
+                // if SRA fails to initialize (the most common reason is a failure to find/load native libraries),
+                // it will throw a subclass of java.lang.Error and here we only catch subclasses of java.lang.Exception
+                //
+                // Note: SRA initialization errors should not be ignored, but rather shown to user
                 return SRAAccession.isValid(sourceFile.getPath());
             } catch (final Exception e) {
```

Comment:
We have such tests for that in ngs-java, but those tests change ngs-java package local variables (and nobody else from outside should touch them) to simulate cases, when libraries are not found locally. Unfortunately, without doing so tests will depend on the environment and cannot be mocked